dashboard/app/models/foorm/library_question.rb

```diff
@@ -22,9 +22,26 @@ class Foorm::LibraryQuestion < ApplicationRecord
   belongs_to :library, primary_key: [:name, :version], foreign_key: [:library_name, :library_version], required: true
 
   validate :validate_library_question
+  validates :question_name, :question, presence: true
 
   after_save :write_to_file
 
+  def published_forms_appeared_in
+    forms_appeared_in = Set.new
+    Foorm::Form.all.each do |form|
```

Comment:
You could check `form.published` first thing here and skip looping through pages if it isn't published. And then remove that extra filtering on line 39